### DISCUSSION TITLE
How to reset FileUpload / SpatieMediaLibraryFileUpload  form field programmatically

### DISCUSSION THREAD

**Discussion by rushabh-indianic** [filamentphp/filament#13307] [Help] [answered]
### Package

Form builder

### Package Version

v3.0.0

### How can we help you?

My application have an admin support ticket system which looks like this
![image](https://github.com/filamentphp/filament/assets/110153864/818fe9c1-358a-4007-9464-b120d66d51ae)

when I send message with attachments, message textarea gets clear but not File Upload field

Now I am using SpatieMediaLibraryFileUpload but same luck with FileUpload

My form handle method looks like this
```php
public function create(): void
{
    $data = $this->form->getState();

    $reply = $this->record->messages()->create([
        'user_id' => auth()->id(),
        'user_type' => get_class(auth()->user()),
        'body' => $data['body'],
        'read_at' => now(),
    ]);

    $this->form->model($reply)->saveRelationships();

    $this->replies->push($reply);

    // Reset form
    $this->form->fill();

    $this->dispatch('new-message');

    // Refresh page to reset form
    $this->redirect('/admin/threads/' . $this->record->id . '/view');
}
```

**Comment by leandrocfe** [comment, marked as answer]
`$this->data['upload_field'] = null`
what happens?

**Reply by rushabh-indianic** [reply]
Works
